### PR TITLE
Test history operations with comps

### PR DESCRIPTION
**Pull request by dmach**
Requires: https://github.com/rpm-software-management/dnf/pull/1689

### CODE REVIEW

File: dnf-behave-tests/features/history-redo-comps.feature [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    And Transaction is following
+        | Action        | Package                           |
+        | remove        | lame-0:3.100-4.fc29.x86_64        |
+        | remove-unused | lame-libs-0:3.100-4.fc29.x86_64   |
+    And History is following
+        | Id     | Command                               | Action        | Altered   |
+        | 4      | history redo 2                        | Removed       | 2         |
+        | 3      | install lame                          | Install       | 2         |
+        | 2      | group remove DNF-CI-Testgroup         | Removed       | 5         |
+        | 1      | group install DNF-CI-Testgroup        | Install       | 5         |
```
[the remark this answers is not in the document]
Comment (lukash) [reply dnf-behave-tests/features/history-redo-comps.feature:53]: This still hasn't been addressed, if you don't want to move the scenarios feel free to state so, but this way I'm not sure it's that or you're just missing my comments.

There still aren't any proper error-handling tests, there's now one for `--skip-unavailable` and printing warnings, but some error handling tests would be really desirable. Adding a few based on my suggestions shouldn't really be that hard...?

File: dnf-behave-tests/features/history-undo.feature [hunk 2]
```diff
@@ -64,3 +64,25 @@ Scenario: Handle missing packages required for undoing the transaction
...
+    Then the exit code is 0
+     And Transaction is following
+         | Action        | Package                      |
+         | upgrade       | flac-0:1.3.3-3.fc29.x86_64   |
+         | upgrade       | wget-0:1.19.6-5.fc29.x86_64  |
+     Then I execute dnf with args "history undo last --skip-unavailable"
+     Then the exit code is 0
+     And Transaction is empty
+     And stderr contains "No package flac-1.3.2-8.fc29.x86_64 available."
+     And stderr contains "No package wget-1.19.5-5.fc29.x86_64 available."
```
Comment (lukash) [inline dnf-behave-tests/features/history-undo.feature:88]: Please use `stderr is ...` with the exact output unless not possible for some reason?

File: dnf-behave-tests/features/transaction-sr/store.feature [hunk 3]
```diff
@@ -58,7 +58,7 @@ Scenario: Store a transaction with an invalid transaction ID
  Then the exit code is 1
   And stderr is
       """
-      Error: Transaction ID "2" not found.
+      Not found given transaction ID
```
Comment (lukash) [inline dnf-behave-tests/features/transaction-sr/store.feature:61]: This is clearly a regression in error message quality :slightly_smiling_face:

File: dnf-behave-tests/features/history-redo-comps.feature [hunk 4]
```diff
@@ -0,0 +1,53 @@
+Feature: Transaction history redo - comps
+
+
+Background:
+  Given I use repository "dnf-ci-fedora"
+    And I use repository "dnf-ci-thirdparty"
+    And I execute dnf with args "group install DNF-CI-Testgroup"
+   Then the exit code is 0
```
[the remark this answers is not in the document]
Comment (lukash) [reply dnf-behave-tests/features/history-redo-comps.feature:8]: Also still not addressed.

File: dnf-behave-tests/features/transaction-sr/store.feature (hunk 3 above)
Comment (dmach) [reply dnf-behave-tests/features/transaction-sr/store.feature]: Fixed

File: dnf-behave-tests/features/steps/transaction.py [hunk 5]
```diff
@@ -117,6 +117,7 @@ def check_dnf_transaction(context, mode):
                 rpm = RPM(nevra)
                 if rpm not in dnf_transaction[action]:
                     candidates = ", ".join([str(i) for i in sorted(dnf_transaction[action])])
+                    print(dnf_transaction)
```
Comment (lukash) [inline dnf-behave-tests/features/steps/transaction.py:120]: Was this left in by accident?

Comment (dmach) [reply dnf-behave-tests/features/steps/transaction.py]: Removed.

**Pull request merged**